bpf/lib/overloadable_skb.h

```diff
@@ -206,10 +208,20 @@ ctx_set_encap_info(struct __sk_buff *ctx, __u32 node_id, __u32 seclabel,
 #endif /* ENABLE_VTEP */
 		key.tunnel_id = seclabel;
 
+	if (src_ip != 0) {
+		key.local_ipv4 = bpf_ntohl(src_ip);
+	} else {
+		/* If we are not setting the outer source IP, we want to pass a smaller
+		 * structure. This is necessary to ensure the helper call doesn't fail
+		 * on older kernels that don't support setting the outer source IP and
+		 * thus expect a smaller structure.
+		 */
+		key_size = offsetof(struct bpf_tunnel_key, local_ipv4);
+	}
```

Comment:
For readability, can we initialize the `key_size` with the smaller/old value and then make this

```suggestion
	if (src_ip != 0) {
		key.local_ipv4 = bpf_ntohl(src_ip);
		key_size = sizeof(key);
	}
```